### PR TITLE
[MM-33908] Resync LDAP Groups to Teams and Channels

### PR DESCRIPTION
**Pull request by maxerenberg**
#### Summary
This PR adds a new parameter to the `/ldap/sync` endpoint which forces members in an LDAP group to join the teams and channels synced to that group, even if they left or were removed.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33908

#### Release Note
```release-note
The `/ldap/sync` endpoint now accepts a parameter `include_removed_members` which will force all LDAP
group members back into the teams and channels synced to that group.
```

#### Enterprise PR
https://github.com/mattermost/enterprise/pull/970

### CODE REVIEW

File: app/app_iface.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ type AppIface interface {
 	CreateDefaultChannels(teamID string) ([]*model.Channel, *model.AppError)
 	// CreateDefaultMemberships adds users to teams and channels based on their group memberships and how those groups
 	// are configured to sync with teams and channels for group members on or after the given timestamp.
-	CreateDefaultMemberships(since int64) error
+	CreateDefaultMemberships(since int64, includeRemovedMembers bool) error
```
Comment (mkraft) [inline app/app_iface.go:83]: Could we please update all of the comments of these interface methods with a clear, simple explanation of what the boolean parameter does?

**Review by mkraft — APPROVED** [review]
@maxerenberg This is really great! Executed perfectly. 👍🏻

**Review by sbishel — APPROVED** [review]
Awesome!

**Comment by sbishel** [comment]
@mkraft - ping on rereview.

**Review by mkraft — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]
Step 1.
image

Step 2.
image

Step 3.User has left Team **B1team**
image

Step 4. syncing using  API
image

Step 5. Team has been added back after Syncing
image

**Pull request merged**